src/main/java/com/sprint/mission/discodeit/security/jwt/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,92 @@
+package com.sprint.mission.discodeit.security.jwt;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sprint.mission.discodeit.dto.data.UserDTO;
+import com.sprint.mission.discodeit.exception.ErrorResponse;
+import com.sprint.mission.discodeit.security.DiscodeitUserDetails;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+  // 헤더 검증 필터
+  private final ObjectMapper objectMapper;
+  private final JwtTokenProvider jwtTokenProvider;
+  private final JwtRegistry<UUID> jwtRegistry;
+
+  @Override
+  protected boolean shouldNotFilter(HttpServletRequest request) throws ServletException {
+    String path = request.getServletPath();
+    return path.equals("/api/auth/refresh") || path.equals("/api/auth/login");
+  }
+
+  @Override
+  protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+      FilterChain filterChain) throws ServletException, IOException {
+    try {
+      String token = resolveToken(request);
+      if (StringUtils.hasText(token)) {
+        if (jwtTokenProvider.validateAccessToken(token) &&
+            jwtRegistry.hasActiveJwtInformationByAccessToken(token)) {
```

Comment:
early return 을 사용하면 좀 깔끔해지지 않을까 싶습니다.
1. 토큰 형식 및 서명 검증 실패시 sendResponse 호출 후 return
2. jwtRegistry.hasActiveJwtInformationByAccessToken 에 실패시 sendResponse 호출 후 return
이후 userDTO 파싱 및 authentication 세팅하는 로직으로 해보면 어떨지?
예시
```suggestion
    if (!jwtTokenProvider.validateAccessToken(token)) {
                sendErrorResponse(response, "Invalid JWT token", HttpServletResponse.SC_UNAUTHORIZED);
                return;
            }

          if (!jwtRegistry.hasActiveJwtInformationByAccessToken(token)) {
           sendErrorResponse(response, "Invalid JWT token", HttpServletResponse.SC_UNAUTHORIZED);
                return;
          }
          ... 이후 로직 생략
```

Reply:
확실히 더 깔끔하네요 보기에도 좋고 개선해놓도록 하겠습니다.